spec/client_server_spec.rb

```diff
@@ -2,6 +2,13 @@
 
 describe EM::Hiredis::Client do
 
+  around(:each) do |test|
+    EM.run do
+      test.run
+      EM.stop
+    end
+  end
```

Comment:
It's not clear to me why these changes require this to be added when it wasn't here previously. Could you explain this please?

Reply:
Because in this implementation I rely on EM timeout function. So if there is no EM in the test it gives you an error:
```RuntimeError:
       eventmachine not initialized: evma_install_oneshot_timer```

Reply:
You can use https://github.com/joshbuddy/em-spec to run tests in an EventMachine context. So in this case you could replace `mock_connections` with

```ruby
  def mock_connections(expected_connections, uri = 'redis://localhost:6379')
    em {

      connectionEM = EM::Hiredis::MockConnectionEM.new(expected_connections, PubsubTestConnection)

      yield EM::Hiredis::PubsubClient.new(uri, nil, nil, connectionEM), connectionEM.connections

      connectionEM.connections.each { |c| c._expectations_met! }

      done
    }
  end
```

I notice that this should also be done for `client_conn_spec.rb` too. The tests for that are currently failing with `RuntimeError: eventmachine not initialized: evma_install_oneshot_timer`.

You have also added

```ruby
  around(:each) do |test|
    EM.run do
      test.run
      EM.stop
    end
  end
```

to `spec/client_server_spec.rb` and `spec/live/redis_commands_spec.rb`, but these are not necessary since these tests are already using `em` where necessary.

Reply:
Thanks! I didn't think of changing `mock_connections`, makes sense.

For this one: `client_conn_spec.rb` i deleted EM today because I've confused this test file with the different one. Returned them back.

For this one `spec/client_server_spec.rb` – it works for me either with `127.0.0.1` instead of `localhost` or by adding
```  around(:each) do |test|
    EM.run do
      test.run
      EM.stop
    end
  end```
Not sure I understand why. I will look into it.